compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -2782,9 +2782,7 @@ private STNode parseFunctionBodyBlock(boolean isAnonFunc) {
                 break;
             }
 
-            //Local type def is not allowed in new spec, hence add it as invalid node minutia.
-            if (stmt.kind == SyntaxKind.LOCAL_TYPE_DEFINITION_STATEMENT) {
-                addInvalidNodeToNextToken(stmt, DiagnosticErrorCode.ERROR_LOCAL_TYPE_DEFINITION_NOT_ALLOWED);
+            if (isStmtNotAllowedInLocalLevel(stmt)) {
```

Comment:
the method name is bit misleading. It looks like just a boolean check, but it also invalidates the node. 
HBT checkAndInvalidateUnallowedStmts?